extensions/iam/oauth2/oauth2-core/src/main/java/org/eclipse/dataspaceconnector/iam/oauth2/core/Oauth2Extension.java

```diff
@@ -108,6 +109,7 @@ public void initialize(ServiceExtensionContext context) {
         var oauth2Service = new Oauth2ServiceImpl(configuration, tokenSigner, okHttpClient, jwtDecoratorRegistry, context.getTypeManager());
 
         context.registerService(IdentityService.class, oauth2Service);
+        context.registerService(Oauth2Service.class, oauth2Service);
```

Comment:
`Oauth2Service` must be declared in the `@Provides` annotation

Reply:
Done (please check again if ok)